### PR TITLE
feat: [CCM-8829]: Rule details page enhancements

### PR DESCRIPTION
**Pull request by rishik30**
### Summary

AutoStopping Rule details page enhancements:
- Implement RBAC  

- Different fields for different rule types  

- Introduce a refresh button for the whole page 

- Convert logs time to local time along with timezone

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

## [Contributor license agreement](https://github.com/harness/harness-core-ui/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 43 Issues 
* Code issue 43 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10960)

File: src/modules/75-ce/common/CERbacToggle.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  disabled,
+  ...props
+}) => {
+  const [canDoAction] = usePermission(permissionRequest, [permissionRequest])
+
+  return (
+    <Popover
+      boundary="viewport"
+      position="top"
+      interactionKind="hover"
```
Comment (jenil-harness) [inline src/modules/75-ce/common/CERbacToggle.tsx:29]: It doesn't support enums from `blueprintjs`?

File: src/modules/75-ce/pages/co-rule-details/CORuleDetailsPage.tsx [hunk 2]
```diff
@@ -145,7 +160,17 @@ const CORuleDetailsPage: React.FC = () => {
                   <Text font={{ variation: FontVariation.H4 }}>{defaultTo(service?.name, '')}</Text>
                   {service && (
                     <Container>
-                      <RuleStatusToggleSwitch serviceData={service} onSuccess={setService} />
+                      <RuleStatusToggleSwitch
+                        serviceData={service}
+                        onSuccess={setService}
+                        enableRbac={true}
```
[the remark this answers is not in the document]
Comment (rishik30) [reply src/modules/75-ce/pages/co-rule-details/CORuleDetailsPage.tsx:166]: this is a prop to enable RBAC or not for the toggle. As CCM has now enabled RBAC completely, so passed in as true

**Comment by rishik30** [comment]
retrigger jest

**Comment by rishik30** [comment]
trigger cypress

**Comment by rishik30** [comment]
trigger cypress

**Review by rajarshee-harness — APPROVED** [review]

**Review by jenil-harness — APPROVED** [review]

**Pull request merged**